src/main/java/slimeknights/tconstruct/library/client/model/BakedSimpleItem.java

```diff
@@ -5,30 +5,75 @@
 
 import java.util.List;
 
-import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.vecmath.Matrix4f;
 
-import slimeknights.mantle.client.model.BakedSimple;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.client.model.ItemLayerModel;
+import net.minecraftforge.client.model.PerspectiveMapWrapper;
+import net.minecraftforge.common.model.TRSRTransformation;
+import org.apache.commons.lang3.tuple.Pair;
 
-public class BakedSimpleItem extends BakedSimple {
+/**
+ * Based on {@link ItemLayerModel.BakedItemModel}
+ * Implements culling for items in the GUI
+ */
+public class BakedSimpleItem implements IBakedModel {
+  private final ImmutableList<BakedQuad> quads;
+  private final TextureAtlasSprite particle;
+  private final ImmutableMap<ItemCameraTransforms.TransformType, TRSRTransformation> transforms;
+  private final IBakedModel otherModel;
+  private final boolean isCulled;
+  private final ItemOverrideList overrides;
 
-  public BakedSimpleItem(ImmutableList<net.minecraft.client.renderer.block.model.BakedQuad> quads, ImmutableMap<net.minecraft.client.renderer.block.model.ItemCameraTransforms.TransformType, net.minecraftforge.common.model.TRSRTransformation> transforms, net.minecraft.client.renderer.block.model.IBakedModel base) {
-    super(quads, transforms, base);
+  public BakedSimpleItem(ImmutableList<BakedQuad> quads, ImmutableMap<ItemCameraTransforms.TransformType, TRSRTransformation> transforms, IBakedModel original) {
+      this(quads, original.getParticleTexture(), transforms, original.getOverrides(), null);
   }
 
-  public BakedSimpleItem(List<net.minecraft.client.renderer.block.model.BakedQuad> quads, ImmutableMap<net.minecraft.client.renderer.block.model.ItemCameraTransforms.TransformType, net.minecraftforge.common.model.TRSRTransformation> transforms, net.minecraft.client.renderer.block.model.IBakedModel base) {
-    super(quads, transforms, base);
+  private BakedSimpleItem(ImmutableList<BakedQuad> quads, TextureAtlasSprite particle, ImmutableMap<ItemCameraTransforms.TransformType, TRSRTransformation> transforms, ItemOverrideList overrides, @Nullable IBakedModel unculledModel) {
```

Comment:
Without looking at forge code.. when is this constructor ever called with a culled model? I couldn't find where it all comes together in the forge PR, kinda feels like something is missing.

Reply:
The code to do this is tricky af, you use the public constructor to create the unculled model.
That constructor calls the private one to create the culled model.
The two are referencing each other, and switch depending on if you are rendering in GUI or not.

Reply:
Yes, that's what I expected it to do. Apparently I overlooked that part in the constructor. Thanks